Implement Window borders

mikkeloscar opened the pull request:
### Implementation

The borders are implemented as a surface/buffer attached to each view
which is sent to and rendered by wlc in the view_pre_render callback.

All the drawing logic is handled in sway/border.c and all the logic for
calculating the geometry of the border/view is handled in
`update_geometry` in sway/layout.c (same place as gaps are calculated).

Included are also support for `border <none|normal|pixel|toggle>` options and `hide_edge_borders <none|vertical|horizontal|both>`.

##### Not included:
Parsing of custom border colors. This needs to be implemented in the config: http://i3wm.org/docs/userguide.html#_changing_colors.

#### Known issues
* [x] The colors are currently dependent on a little-endian cpu to be correct (discussing [here](https://github.com/Cloudef/wlc/issues/142#issuecomment-202886577)). This needs to be corrected before merge. I suggest we just do a runtime check of endianness and format the colors based on that unless someone has a better idea.
* Moving a floating window off the edge off the screen will confuse the border rendering. It's because wlc clamps the geometry, so we just need to send the right part of the border in those cases (haven't bothered with this yet).
* [x] The title does not change. I don't know if wlc provides a way to get the current window title, haven't looked into it.
* Borders on current windows does not change if you reload the config with a new border config. (Just need to check all views and update those needed, haven't bothered with this yet either).


#### TODO

* [x] Isn't there supposed to be a different border for windows that are focused, but on an output that is not focused?
* [x] My font is incorrect now for swaybar. Oddly it's correct for the window titles.
* [x] Another issue: you can see the border through transparent surfaces.
* [x] A runtime check on endianness works for me.
* [x] Commented code is bad. Why is this commented? It's kind of important, probably related to why gnome-calculator is fucked up.

SirCmpwn commented:
Very cool stuff. Some feedback before I dive into the code:

1. Isn't there supposed to be a different border for windows that are focused, but on an output that is not focused?
2. My font is incorrect now for swaybar. Oddly it's correct for the window titles.

A runtime check on endianness works for me.

SirCmpwn commented:
Another issue: you can see the border through transparent surfaces.

SirCmpwn commented:
Mar 29 13:27:15 (process:6942): Pango-CRITICAL **: pango_font_description_from_string: assertion 'str != NULL' failed

From my log. 6942 is my swaybar process.

SirCmpwn commented:
gnome-calculator freaks out when I'm running this branch, too.

SirCmpwn commented on sway/border.c:
```diff
@@ -0,0 +1,188 @@
+	cairo_t *cr;
+	view->border_geometry = geo;
+	int stride = cairo_format_stride_for_width(CAIRO_FORMAT_ARGB32, geo.size.w);
+	view->border = calloc(stride * geo.size.h, sizeof(unsigned char));
```
Are we leaking memory every time the border updates? Where do we free this?

SirCmpwn commented on sway/handlers.c:
```diff
@@ -344,13 +349,13 @@ static void handle_view_geometry_request(wlc_handle handle, const struct wlc_geo
+		/* 	view->x = geometry->origin.x; */
+		/* 	view->y = geometry->origin.y; */
+		/* /1* 	arrange_windows(view->parent, -1, -1); *1/ */
+		/* } */
```
Commented code is bad. Why is this commented? It's kind of important, probably related to why gnome-terminal is fucked up.

SirCmpwn replied:
Note: gnome-calculator is made to float in my config file with a for_window rule.

mikkeloscar replied:
It's commented out for that reason, I didn't know what it might break to remove it. The issue is that arrange_windows calls wlc_view_set_geometry at some point and a call to `wlc_view_set_geometry` triggers the `handle_view_geometry_request`, so you end up with the bug described in: #487 and this bug appeared with all floating windows with the border code.

Now I know what it might break, so I can investigate. thanks :)

mikkeloscar replied:
@SirCmpwn how does gnome-calculator freak out? It seems to work just fine when I make it floating. Also tried with your config, though I don't see a for_window rule in the config you linked on irc.

SirCmpwn replied:
Sorry, the config I linked on IRC is actually a different one than I was testing with at the time.

    for_window [title="^gnome-calculator$"] floating enable

It freaks out by appearing briefly and then becoming very small, I think. I can still move up to the floating layer but I don't see it and I can't interact with it. I can also close it (with keybindings).

mikkeloscar replied:
Ok, when spawned like that I also see it. It's definitely linked to this commented out code, and the border geometry logic. Also related to the firefox crash.

SirCmpwn commented:
That's all I've got. Looks great, nice work!

SirCmpwn commented:
>The title does not change. I don't know if wlc provides a way to get the current window title, haven't looked into it.

Same way we get it in the first place, if I recall correctly.

    /** Get title. */
    const char* wlc_view_get_title(wlc_handle view);

SirCmpwn commented:
I get a crash on this branch when initiating a download in firefox.

    [main.c:39] [wlc] xwm: function set_geometry at line 307 x11 error code 11
    [handlers.c:347] geometry request for 7 499x379 @ 1261,706
    [main.c:39] [wlc] xwm: function set_geometry at line 307 x11 error code 11
    sway: /tmp/yaourt-tmp-sircmpwn/aur-wlc-git/src/wlc/src/compositor/output.c:142: blit: Assertion `b->x - a->x >= 0' failed.
    [log.c:150] Error: Signal 6. Printing backtrace
    [log.c:160] Backtrace: sway(error_handler+0xb7) [0x426aaf]
    [log.c:160] Backtrace: /usr/lib/libc.so.6(+0x33330) [0x7fd659773330]
    [log.c:160] Backtrace: /usr/lib/libc.so.6(gsignal+0x38) [0x7fd6597732a8]
    [log.c:160] Backtrace: /usr/lib/libc.so.6(abort+0x16a) [0x7fd65977472a]
    [log.c:160] Backtrace: /usr/lib/libc.so.6(+0x2c1b7) [0x7fd65976c1b7]
    [log.c:160] Backtrace: /usr/lib/libc.so.6(+0x2c262) [0x7fd65976c262]
    [log.c:160] Backtrace: /usr/lib/libwlc.so.0(+0x22190) [0x7fd65b80d190]
    [log.c:160] Backtrace: /usr/lib/libwayland-server.so.0(+0x9157) [0x7fd65af27157]
    [log.c:160] Backtrace: /usr/lib/libwayland-server.so.0(wl_event_loop_dispatch+0x72) [0x7fd65af27832]
    [log.c:160] Backtrace: /usr/lib/libwayland-server.so.0(wl_display_run+0x25) [0x7fd65af25ee5]
    [log.c:160] Backtrace: /usr/lib/libwlc.so.0(wlc_run+0xec) [0x7fd65b81682c]
    [log.c:160] Backtrace: sway(main+0x473) [0x422beb]
    [log.c:160] Backtrace: /usr/lib/libc.so.6(__libc_start_main+0xf0) [0x7fd659760710]
    [log.c:160] Backtrace: sway(_start+0x29) [0x407fc9]
    [log.c:163] Maps:
    [log.c:174] 00400000-00432000 r-xp 00000000 fe:00 1866366                            /usr/bin/sway
    [log.c:174] 00631000-00633000 rw-p 00031000 fe:00 1866366                            /usr/bin/sway
    [log.c:174] 016b4000-040ac000 rw-p 00000000 00:00 0                                  [heap]
    [log.c:174] 7fd647ef8000-7fd6496f8000 rw-s 00000000 00:05 299620                     /drm mm object (deleted)
    [log.c:174] 7fd6496f8000-7fd64a4f8000 rw-s 00000000 00:05 299626                     /drm mm object (deleted)
    [log.c:174] 7fd64a4f8000-7fd64bcf8000 rw-s 00000000 00:05 285420                     /drm mm object (deleted)
    [log.c:174] 7fd64bcf8000-7fd64bdf7000 rw-s 00000000 00:2c 298881                     /run/user/1000/weston-shared-XWFJr8 (deleted)
    [log.c:174] 7fd64bdf7000-7fd64cc07000 rw-s 00000000 00:2c 298880                     /run/user/1000/sway-client-tuMlBd (deleted)
    [log.c:174] 7fd64cc07000-7fd64cc87000 rw-s 00000000 00:05 285418                     /drm mm object (deleted)
    [log.c:174] 7fd64cc87000-7fd64da87000 rw-s 00000000 00:05 285417                     /drm mm object (deleted)
    [log.c:174] 7fd64da87000-7fd64db87000 rw-s 00000000 00:05 285416                     /drm mm object (deleted)
    [log.c:174] 7fd64db87000-7fd64dc86000 rw-s 00000000 00:2c 294569                     /run/user/1000/weston-shared-j5Qwr8 (deleted)
    [log.c:174] 7fd64dc86000-7fd64e550000 rw-s 00000000 00:2c 294568                     /run/user/1000/sway-client-u3w8Ad (deleted)
    [log.c:174] 7fd64e550000-7fd64e64f000 rw-s 00000000 00:2c 297864                     /run/user/1000/weston-shared-nWarHO (deleted)
    [log.c:174] 7fd64e6ce000-7fd64eece000 rw-s 00000000 00:05 280477                     /drm mm object (deleted)
    [log.c:174] 7fd64eece000-7fd64f3c0000 rw-s 00000000 00:2c 297858                     /run/user/1000/sway-client-igWMqa (deleted)
    [log.c:174] 7fd64f3c0000-7fd64f3c4000 r-xp 00000000 fe:00 1861464                    /usr/lib/libtxc_dxtn.so
    [log.c:174] 7fd64f3c4000-7fd64f5c3000 ---p 00004000 fe:00 1861464                    /usr/lib/libtxc_dxtn.so
    [log.c:174] 7fd64f5c3000-7fd64f5c4000 r--p 00003000 fe:00 1861464                    /usr/lib/libtxc_dxtn.so
    [log.c:174] 7fd64f5c4000-7fd64f5c5000 rw-p 00004000 fe:00 1861464                    /usr/lib/libtxc_dxtn.so
    [log.c:174] 7fd64f5f5000-7fd64f5fd000 rw-s 00000000 00:05 299582                     /drm mm object (deleted)
    [log.c:174] 7fd64f5fd000-7fd64f605000 rw-s 00000000 00:05 299581                     /drm mm object (deleted)
    [log.c:174] 7fd64f605000-7fd64f62c000 rw-p 00000000 00:00 0 
    [log.c:174] 7fd64f667000-7fd64f766000 rw-s 00000000 00:2c 297861                     /run/user/1000/weston-shared-nCyFvY (deleted)
    [log.c:174] 7fd64f766000-7fd64f865000 rw-s 00000000 00:2c 280471                     /run/user/1000/weston-shared-9ZL0m8 (deleted)
    [log.c:174] 7fd64f8ca000-7fd64f8cb000 rw-p 00000000 00:00 0 
    [log.c:174] 7fd64f8cb000-7fd64f8cc000 rw-s 00000000 00:2c 300579                     /run/user/1000/xwayland-shared-9dnrvc (deleted)
    [log.c:174] 7fd64f8cc000-7fd64f8cd000 rw-s 00000000 00:05 299628                     /drm mm object (deleted)
    [log.c:174] 7fd64f8cd000-7fd64f8ce000 rw-s 00000000 00:05 300400                     /drm mm object (deleted)
    [log.c:174] 7fd64f8ce000-7fd64f8cf000 rw-s 00000000 00:2c 298938                     /run/user/1000/xwayland-shared-Ze8tSZ (deleted)
    [log.c:174] 7fd64f8cf000-7fd64f8d1000 rw-s 00000000 00:05 297983                     /drm mm object (deleted)
    [log.c:174] 7fd64f8d1000-7fd64f8d2000 rw-s 00000000 00:2c 300418                     /run/user/1000/xwayland-shared-tMgCXa (deleted)
    [log.c:174] 7fd64f8d2000-7fd64f8d3000 rw-s 00000000 00:05 299617                     /drm mm object (deleted)
    [log.c:174] 7fd64f8d3000-7fd64f8d4000 rw-s 00000000 00:2c 297871                     /run/user/1000/xwayland-shared-LVtleh (deleted)
    [log.c:174] 7fd64f8d4000-7fd64f8e4000 rw-s 00000000 00:05 300399                     /drm mm object (deleted)
    [log.c:174] 7fd64f8e4000-7fd64f9e3000 rw-s 00000000 00:2c 297859                     /run/user/1000/weston-shared-Dq1Yg5 (deleted)
    [log.c:174] 7fd64f9e3000-7fd64f9f3000 rw-s 00000000 00:05 285421                     /drm mm object (deleted)
    [log.c:174] 7fd64f9f3000-7fd64fa53000 rw-s 00000000 00:05 285419                     /drm mm object (deleted)
    [log.c:174] 7fd64fa53000-7fd64fa7f000 rw-s 00000000 00:2c 297868                     /run/user/1000/sway-client-xzXa3z (deleted)
    [log.c:174] 7fd64fa7f000-7fd64fab9000 rw-s 00000000 00:2c 297867                     /run/user/1000/sway-client-HPJSVE (deleted)
    [log.c:174] 7fd64fab9000-7fd64fada000 rw-s 00000000 00:2c 297866                     /run/user/1000/sway-client-P2yCOJ (deleted)
    [log.c:174] 7fd64fada000-7fd64fae2000 rw-s 00000000 00:05 299629                     /drm mm object (deleted)
    [log.c:174] 7fd64fae2000-7fd64faea000 rw-s 00000000 00:05 299623                     /drm mm object (deleted)
    [log.c:174] 7fd64faea000-7fd64fafa000 rw-s 00000000 00:05 299618                     /drm mm object (deleted)
    [log.c:174] 7fd64fafa000-7fd64fafb000 rw-s 00000000 00:05 299613                     /drm mm object (deleted)
    [log.c:174] 7fd64fafb000-7fd64fafc000 rw-s 00000000 00:05 299611                     /drm mm object (deleted)
    [log.c:174] 7fd64fafc000-7fd64fb04000 rw-s 00000000 00:05 299610                     /drm mm object (deleted)
    [log.c:174] 7fd64fb04000-7fd64fb05000 rw-s 100002000 00:06 13742                     /dev/dri/card0
    [log.c:174] 7fd64fb05000-7fd64fb0d000 rw-s 00000000 00:05 299603                     /drm mm object (deleted)
    [log.c:174] 7fd64fb0d000-7fd64fb0e000 rw-s 00000000 00:05 299600                     /drm mm object (deleted)
    [log.c:174] 7fd64fb0e000-7fd64fb16000 rw-s 00000000 00:05 299597                     /drm mm object (deleted)
    [log.c:174] 7fd64fb16000-7fd64fb1e000 rw-s 00000000 00:05 299590                     /drm mm object (deleted)
    [log.c:174] 7fd64fb1e000-7fd64fb72000 r--p 00000000 fe:00 406459                     /usr/share/fonts/ttf-dejavu-ib/DejaVuSansMono.ttf
    [log.c:174] 7fd64fb72000-7fd64fb9a000 rw-p 00000000 00:00 0 
    [log.c:174] 7fd64fb9a000-7fd64fba2000 r-xp 00000000 fe:00 1857855                    /usr/lib/libpciaccess.so.0.11.1
    [log.c:174] 7fd64fba2000-7fd64fda2000 ---p 00008000 fe:00 1857855                    /usr/lib/libpciaccess.so.0.11.1
    [log.c:174] 7fd64fda2000-7fd64fda3000 r--p 00008000 fe:00 1857855                    /usr/lib/libpciaccess.so.0.11.1
    [log.c:174] 7fd64fda3000-7fd64fda4000 rw-p 00009000 fe:00 1857855                    /usr/lib/libpciaccess.so.0.11.1
    [log.c:174] 7fd64fda4000-7fd64fdba000 r-xp 00000000 fe:00 1838523                    /usr/lib/libgcc_s.so.1
    [log.c:174] 7fd64fdba000-7fd64ffb9000 ---p 00016000 fe:00 1838523                    /usr/lib/libgcc_s.so.1
    [log.c:174] 7fd64ffb9000-7fd64ffba000 rw-p 00015000 fe:00 1838523                    /usr/lib/libgcc_s.so.1
    [log.c:174] 7fd64ffba000-7fd65012c000 r-xp 00000000 fe:00 1838568                    /usr/lib/libstdc++.so.6.0.21
    [log.c:174] 7fd65012c000-7fd65032c000 ---p 00172000 fe:00 1838568                    /usr/lib/libstdc++.so.6.0.21
    [log.c:174] 7fd65032c000-7fd650336000 r--p 00172000 fe:00 1838568                    /usr/lib/libstdc++.so.6.0.21
    [log.c:174] 7fd650336000-7fd650338000 rw-p 0017c000 fe:00 1838568                    /usr/lib/libstdc++.so.6.0.21
    [log.c:174] 7fd650338000-7fd65033c000 rw-p 00000000 00:00 0 
    [log.c:174] 7fd65033c000-7fd650347000 r-xp 00000000 fe:00 1888741                    /usr/lib/libdrm_radeon.so.1.0.1
    [log.c:174] 7fd650347000-7fd650546000 ---p 0000b000 fe:00 1888741                    /usr/lib/libdrm_radeon.so.1.0.1
    [log.c:174] 7fd650546000-7fd650547000 r--p 0000a000 fe:00 1888741                    /usr/lib/libdrm_radeon.so.1.0.1
    [log.c:174] 7fd650547000-7fd650548000 rw-p 0000b000 fe:00 1888741                    /usr/lib/libdrm_radeon.so.1.0.1
    [log.c:174] 7fd650548000-7fd65054f000 r-xp 00000000 fe:00 1888750                    /usr/lib/libdrm_nouveau.so.2.0.0
    [log.c:174] 7fd65054f000-7fd65074e000 ---p 00007000 fe:00 1888750                    /usr/lib/libdrm_nouveau.so.2.0.0
    [log.c:174] 7fd65074e000-7fd65074f000 r--p 00006000 fe:00 1888750                    /usr/lib/libdrm_nouveau.so.2.0.0
    [log.c:174] 7fd65074f000-7fd650750000 rw-p 00007000 fe:00 1888750                    /usr/lib/libdrm_nouveau.so.2.0.0
    [log.c:174] 7fd650750000-7fd650770000 r-xp 00000000 fe:00 1888742                    /usr/lib/libdrm_intel.so.1.0.0
    [log.c:174] 7fd650770000-7fd65096f000 ---p 00020000 fe:00 1888742                    /usr/lib/libdrm_intel.so.1.0.0
    [log.c:174] 7fd65096f000-7fd650970000 r--p 0001f000 fe:00 1888742                    /usr/lib/libdrm_intel.so.1.0.0
    [log.c:174] 7fd650970000-7fd650971000 rw-p 00020000 fe:00 1888742                    /usr/lib/libdrm_intel.so.1.0.0
    [log.c:174] 7fd650971000-7fd650f49000 r-xp 00000000 fe:00 8313                       /usr/lib/xorg/modules/dri/i965_dri.so
    [log.c:174] 7fd650f49000-7fd651148000 ---p 005d8000 fe:00 8313                       /usr/lib/xorg/modules/dri/i965_dri.so
    [log.c:174] 7fd651148000-7fd651179000 r--p 005d7000 fe:00 8313                       /usr/lib/xorg/modules/dri/i965_dri.so
    [log.c:174] 7fd651179000-7fd651180000 rw-p 00608000 fe:00 8313                       /usr/lib/xorg/modules/dri/i965_dri.so
    [log.c:174] 7fd651180000-7fd651188000 rw-p 00000000 00:00 0 
    [log.c:174] 7fd651188000-7fd65118d000 r-xp 00000000 fe:00 1861327                    /usr/lib/libXxf86vm.so.1.0.0
    [log.c:174] 7fd65118d000-7fd65138c000 ---p 00005000 fe:00 1861327                    /usr/lib/libXxf86vm.so.1.0.0
    [log.c:174] 7fd65138c000-7fd65138d000 r--p 00004000 fe:00 1861327                    /usr/lib/libXxf86vm.so.1.0.0
    [log.c:174] 7fd65138d000-7fd65138e000 rw-p 00005000 fe:00 1861327                    /usr/lib/libXxf86vm.so.1.0.0
    [log.c:174] 7fd65138e000-7fd6513a5000 r-xp 00000000 fe:00 1860321                    /usr/lib/libxcb-glx.so.0.0.0
    [log.c:174] 7fd6513a5000-7fd6515a5000 ---p 00017000 fe:00 1860321                    /usr/lib/libxcb-glx.so.0.0.0
    [log.c:174] 7fd6515a5000-7fd6515a7000 r--p 00017000 fe:00 1860321                    /usr/lib/libxcb-glx.so.0.0.0
    [log.c:174] 7fd6515a7000-7fd6515a8000 rw-p 00019000 fe:00 1860321                    /usr/lib/libxcb-glx.so.0.0.0
    [log.c:174] 7fd6515a8000-7fd6515ad000 r-xp 00000000 fe:00 1861332                    /usr/lib/libXfixes.so.3.1.0
    [log.c:174] 7fd6515ad000-7fd6517ac000 ---p 00005000 fe:00 1861332                    /usr/lib/libXfixes.so.3.1.0
    [log.c:174] 7fd6517ac000-7fd6517ad000 r--p 00004000 fe:00 1861332                    /usr/lib/libXfixes.so.3.1.0
    [log.c:174] 7fd6517ad000-7fd6517ae000 rw-p 00005000 fe:00 1861332                    /usr/lib/libXfixes.so.3.1.0
    [log.c:174] 7fd6517ae000-7fd6517b0000 r-xp 00000000 fe:00 1861338                    /usr/lib/libXdamage.so.1.1.0
    [log.c:174] 7fd6517b0000-7fd6519af000 ---p 00002000 fe:00 1861338                    /usr/lib/libXdamage.so.1.1.0
    [log.c:174] 7fd6519af000-7fd6519b0000 r--p 00001000 fe:00 1861338                    /usr/lib/libXdamage.so.1.1.0
    [log.c:174] 7fd6519b0000-7fd6519b1000 rw-p 00002000 fe:00 1861338                    /usr/lib/libXdamage.so.1.1.0
    [log.c:174] 7fd6519b1000-7fd6519c0000 r-xp 00000000 fe:00 1841274                    /usr/lib/libbz2.so.1.0.6
    [log.c:174] 7fd6519c0000-7fd651bbf000 ---p 0000f000 fe:00 1841274                    /usr/lib/libbz2.so.1.0.6
    [log.c:174] 7fd651bbf000-7fd651bc1000 rw-p 0000e000 fe:00 1841274                    /usr/lib/libbz2.so.1.0.6
    [log.c:174] 7fd651bc1000-7fd651bea000 r-xp 00000000 fe:00 1872899                    /usr/lib/libgraphite2.so.3.0.1
    [log.c:174] 7fd651bea000-7fd651dea000 ---p 00029000 fe:00 1872899                    /usr/lib/libgraphite2.so.3.0.1
    [log.c:174] 7fd651dea000-7fd651dec000 r--p 00029000 fe:00 1872899                    /usr/lib/libgraphite2.so.3.0.1
    [log.c:174] 7fd651dec000-7fd651ded000 rw-p 0002b000 fe:00 1872899                    /usr/lib/libgraphite2.so.3.0.1
    [log.c:174] 7fd651ded000-7fd651df4000 r-xp 00000000 fe:00 1873565                    /usr/lib/libdatrie.so.1.3.3
    [log.c:174] 7fd651df4000-7fd651ff3000 ---p 00007000 fe:00 1873565                    /usr/lib/libdatrie.so.1.3.3
    [log.c:174] 7fd651ff3000-7fd651ff4000 r--p 00006000 fe:00 1873565                    /usr/lib/libdatrie.so.1.3.3
    [log.c:174] 7fd651ff4000-7fd651ff5000 rw-p 00007000 fe:00 1873565                    /usr/lib/libdatrie.so.1.3.3
    [log.c:174] 7fd651ff5000-7fd651ffe000 r-xp 00000000 fe:00 1861613                    /usr/lib/libgudev-1.0.so.0.2.0
    [log.c:174] 7fd651ffe000-7fd6521fd000 ---p 00009000 fe:00 1861613                    /usr/lib/libgudev-1.0.so.0.2.0
    [log.c:174] 7fd6521fd000-7fd6521fe000 r--p 00008000 fe:00 1861613                    /usr/lib/libgudev-1.0.so.0.2.0
    [log.c:174] 7fd6521fe000-7fd6521ff000 rw-p 00009000 fe:00 1861613                    /usr/lib/libgudev-1.0.so.0.2.0
    [log.c:174] 7fd6521ff000-7fd652212000 r-xp 00000000 fe:00 1866752                    /usr/lib/libgpg-error.so.0.17.0
    [log.c:174] 7fd652212000-7fd652411000 ---p 00013000 fe:00 1866752                    /usr/lib/libgpg-error.so.0.17.0
    [log.c:174] 7fd652411000-7fd652412000 r--p 00012000 fe:00 1866752                    /usr/lib/libgpg-error.so.0.17.0
    [log.c:174] 7fd652412000-7fd652413000 rw-p 00013000 fe:00 1866752                    /usr/lib/libgpg-error.so.0.17.0
    [log.c:174] 7fd652413000-7fd6524eb000 r-xp 00000000 fe:00 1846502                    /usr/lib/libgcrypt.so.20.0.5
    [log.c:174] 7fd6524eb000-7fd6526eb000 ---p 000d8000 fe:00 1846502                    /usr/lib/libgcrypt.so.20.0.5
    [log.c:174] 7fd6526eb000-7fd6526ec000 r--p 000d8000 fe:00 1846502                    /usr/lib/libgcrypt.so.20.0.5
    [log.c:174] 7fd6526ec000-7fd6526f4000 rw-p 000d9000 fe:00 1846502                    /usr/lib/libgcrypt.so.20.0.5
    [log.c:174] 7fd6526f4000-7fd6526f5000 rw-p 00000000 00:00 0 
    [log.c:174] 7fd6526f5000-7fd652706000 r-xp 00000000 fe:00 1846516                    /usr/lib/liblz4.so.1.7.1
    [log.c:174] 7fd652706000-7fd652905000 ---p 00011000 fe:00 1846516                    /usr/lib/liblz4.so.1.7.1
    [log.c:174] 7fd652905000-7fd652906000 r--p 00010000 fe:00 1846516                    /usr/lib/liblz4.so.1.7.1
    [log.c:174] 7fd652906000-7fd652907000 rw-p 00011000 fe:00 1846516                    /usr/lib/liblz4.so.1.7.1
    [log.c:174] 7fd652907000-7fd65292c000 r-xp 00000000 fe:00 1846387                    /usr/lib/liblzma.so.5.2.2
    [log.c:174] 7fd65292c000-7fd652b2b000 ---p 00025000 fe:00 1846387                    /usr/lib/liblzma.so.5.2.2
    [log.c:174] 7fd652b2b000-7fd652b2c000 r--p 00024000 fe:00 1846387                    /usr/lib/liblzma.so.5.2.2
    [log.c:174] 7fd652b2c000-7fd652b2d000 rw-p 00025000 fe:00 1846387                    /usr/lib/liblzma.so.5.2.2
    [log.c:174] 7fd652b2d000-7fd652b54000 r-xp 00000000 fe:00 1911915                    /usr/lib/libexpat.so.1.6.0
    [log.c:174] 7fd652b54000-7fd652d54000 ---p 00027000 fe:00 1911915                    /usr/lib/libexpat.so.1.6.0
    [log.c:174] 7fd652d54000-7fd652d56000 r--p 00027000 fe:00 1911915                    /usr/lib/libexpat.so.1.6.0
    [log.c:174] 7fd652d56000-7fd652d57000 rw-p 00029000 fe:00 1911915                    /usr/lib/libexpat.so.1.6.0
    [log.c:174] 7fd652d57000-7fd652d63000 r-xp 00000000 fe:00 1857923                    /usr/lib/libwayland-client.so.0.3.0
    [log.c:174] 7fd652d63000-7fd652f63000 ---p 0000c000 fe:00 1857923                    /usr/lib/libwayland-client.so.0.3.0
    [log.c:174] 7fd652f63000-7fd652f65000 r--p 0000c000 fe:00 1857923                    /usr/lib/libwayland-client.so.0.3.0
    [log.c:174] 7fd652f65000-7fd652f66000 rw-p 0000e000 fe:00 1857923                    /usr/lib/libwayland-client.so.0.3.0
    [log.c:174] 7fd652f66000-7fd652f67000 r-xp 00000000 fe:00 1861342                    /usr/lib/libxshmfence.so.1.0.0
    [log.c:174] 7fd652f67000-7fd653167000 ---p 00001000 fe:00 1861342                    /usr/lib/libxshmfence.so.1.0.0
    [log.c:174] 7fd653167000-7fd653168000 r--p 00001000 fe:00 1861342                    /usr/lib/libxshmfence.so.1.0.0
    [log.c:174] 7fd653168000-7fd653169000 rw-p 00002000 fe:00 1861342                    /usr/lib/libxshmfence.so.1.0.0
    [log.c:174] 7fd653169000-7fd65316e000 r-xp 00000000 fe:00 1860371                    /usr/lib/libxcb-sync.so.1.0.0
    [log.c:174] 7fd65316e000-7fd65336e000 ---p 00005000 fe:00 1860371                    /usr/lib/libxcb-sync.so.1.0.0
    [log.c:174] 7fd65336e000-7fd65336f000 r--p 00005000 fe:00 1860371                    /usr/lib/libxcb-sync.so.1.0.0
    [log.c:174] 7fd65336f000-7fd653370000 rw-p 00006000 fe:00 1860371                    /usr/lib/libxcb-sync.so.1.0.0
    [log.c:174] 7fd653370000-7fd653373000 r-xp 00000000 fe:00 1860334                    /usr/lib/libxcb-shape.so.0.0.0
    [log.c:174] 7fd653373000-7fd653572000 ---p 00003000 fe:00 1860334                    /usr/lib/libxcb-shape.so.0.0.0
    [log.c:174] 7fd653572000-7fd653573000 r--p 00002000 fe:00 1860334                    /usr/lib/libxcb-shape.so.0.0.0
    [log.c:174] 7fd653573000-7fd653574000 rw-p 00003000 fe:00 1860334                    /usr/lib/libxcb-shape.so.0.0.0
    [log.c:174] 7fd653574000-7fd653581000 r-xp 00000000 fe:00 1860379                    /usr/lib/libxcb-randr.so.0.1.0
    [log.c:174] 7fd653581000-7fd653780000 ---p 0000d000 fe:00 1860379                    /usr/lib/libxcb-randr.so.0.1.0
    [log.c:174] 7fd653780000-7fd653781000 r--p 0000c000 fe:00 1860379                    /usr/lib/libxcb-randr.so.0.1.0
    [log.c:174] 7fd653781000-7fd653782000 rw-p 0000d000 fe:00 1860379                    /usr/lib/libxcb-randr.so.0.1.0
    [log.c:174] 7fd653782000-7fd653784000 r-xp 00000000 fe:00 1860356                    /usr/lib/libxcb-present.so.0.0.0
    [log.c:174] 7fd653784000-7fd653983000 ---p 00002000 fe:00 1860356                    /usr/lib/libxcb-present.so.0.0.0
    [log.c:174] 7fd653983000-7fd653984000 r--p 00001000 fe:00 1860356                    /usr/lib/libxcb-present.so.0.0.0
    [log.c:174] 7fd653984000-7fd653985000 rw-p 00002000 fe:00 1860356                    /usr/lib/libxcb-present.so.0.0.0
    [log.c:174] 7fd653985000-7fd653987000 r-xp 00000000 fe:00 1860322                    /usr/lib/libxcb-dri3.so.0.0.0
    [log.c:174] 7fd653987000-7fd653b86000 ---p 00002000 fe:00 1860322                    /usr/lib/libxcb-dri3.so.0.0.0
    [log.c:174] 7fd653b86000-7fd653b87000 r--p 00001000 fe:00 1860322                    /usr/lib/libxcb-dri3.so.0.0.0
    [log.c:174] 7fd653b87000-7fd653b88000 rw-p 00002000 fe:00 1860322                    /usr/lib/libxcb-dri3.so.0.0.0
    [log.c:174] 7fd653b88000-7fd653b8c000 r-xp 00000000 fe:00 1860353                    /usr/lib/libxcb-dri2.so.0.0.0
    [log.c:174] 7fd653b8c000-7fd653d8b000 ---p 00004000 fe:00 1860353                    /usr/lib/libxcb-dri2.so.0.0.0
    [log.c:174] 7fd653d8b000-7fd653d8c000 r--p 00003000 fe:00 1860353                    /usr/lib/libxcb-dri2.so.0.0.0
    [log.c:174] 7fd653d8c000-7fd653d8d000 rw-p 00004000 fe:00 1860353                    /usr/lib/libxcb-dri2.so.0.0.0
    [log.c:174] 7fd653d8d000-7fd653db6000 r-xp 00000000 fe:00 1861530                    /usr/lib/libglapi.so.0.0.0
    [log.c:174] 7fd653db6000-7fd653fb5000 ---p 00029000 fe:00 1861530                    /usr/lib/libglapi.so.0.0.0
    [log.c:174] 7fd653fb5000-7fd653fb9000 r--p 00028000 fe:00 1861530                    /usr/lib/libglapi.so.0.0.0
    [log.c:174] 7fd653fb9000-7fd653fba000 rw-p 0002c000 fe:00 1861530                    /usr/lib/libglapi.so.0.0.0
    [log.c:174] 7fd653fba000-7fd653fbb000 rw-p 00000000 00:00 0 
    [log.c:174] 7fd653fbb000-7fd653fc0000 r-xp 00000000 fe:00 1861593                    /usr/lib/libxcb-util.so.1.0.0
    [log.c:174] 7fd653fc0000-7fd6541bf000 ---p 00005000 fe:00 1861593                    /usr/lib/libxcb-util.so.1.0.0
    [log.c:174] 7fd6541bf000-7fd6541c0000 r--p 00004000 fe:00 1861593                    /usr/lib/libxcb-util.so.1.0.0
    [log.c:174] 7fd6541c0000-7fd6541c1000 rw-p 00005000 fe:00 1861593                    /usr/lib/libxcb-util.so.1.0.0
    [log.c:174] 7fd6541c1000-7fd6541c6000 r-xp 00000000 fe:00 1858037                    /usr/lib/libXdmcp.so.6.0.0
    [log.c:174] 7fd6541c6000-7fd6543c5000 ---p 00005000 fe:00 1858037                    /usr/lib/libXdmcp.so.6.0.0
    [log.c:174] 7fd6543c5000-7fd6543c6000 r--p 00004000 fe:00 1858037                    /usr/lib/libXdmcp.so.6.0.0
    [log.c:174] 7fd6543c6000-7fd6543c7000 rw-p 00005000 fe:00 1858037                    /usr/lib/libXdmcp.so.6.0.0
    [log.c:174] 7fd6543c7000-7fd6543c9000 r-xp 00000000 fe:00 1858041                    /usr/lib/libXau.so.6.0.0
    [log.c:174] 7fd6543c9000-7fd6545c9000 ---p 00002000 fe:00 1858041                    /usr/lib/libXau.so.6.0.0
    [log.c:174] 7fd6545c9000-7fd6545ca000 r--p 00002000 fe:00 1858041                    /usr/lib/libXau.so.6.0.0
    [log.c:174] 7fd6545ca000-7fd6545cb000 rw-p 00003000 fe:00 1858041                    /usr/lib/libXau.so.6.0.0
    [log.c:174] 7fd6545cb000-7fd6545df000 r-xp 00000000 fe:00 1838314                    /usr/lib/libresolv-2.23.so
    [log.c:174] 7fd6545df000-7fd6547de000 ---p 00014000 fe:00 1838314                    /usr/lib/libresolv-2.23.so
    [log.c:174] 7fd6547de000-7fd6547df000 r--p 00013000 fe:00 1838314                    /usr/lib/libresolv-2.23.so
    [log.c:174] 7fd6547df000-7fd6547e0000 rw-p 00014000 fe:00 1838314                    /usr/lib/libresolv-2.23.so
    [log.c:174] 7fd6547e0000-7fd6547e2000 rw-p 00000000 00:00 0 
    [log.c:174] 7fd6547e2000-7fd6547e6000 r-xp 00000000 fe:00 1841393                    /usr/lib/libcap.so.2.25
    [log.c:174] 7fd6547e6000-7fd6549e5000 ---p 00004000 fe:00 1841393                    /usr/lib/libcap.so.2.25
    [log.c:174] 7fd6549e5000-7fd6549e6000 rw-p 00003000 fe:00 1841393                    /usr/lib/libcap.so.2.25
    [log.c:174] 7fd6549e6000-7fd654b5e000 r-xp 00000000 fe:00 1848371                    /usr/lib/libgio-2.0.so.0.4600.2
    [log.c:174] 7fd654b5e000-7fd654d5d000 ---p 00178000 fe:00 1848371                    /usr/lib/libgio-2.0.so.0.4600.2
    [log.c:174] 7fd654d5d000-7fd654d61000 r--p 00177000 fe:00 1848371                    /usr/lib/libgio-2.0.so.0.4600.2
    [log.c:174] 7fd654d61000-7fd654d63000 rw-p 0017b000 fe:00 1848371                    /usr/lib/libgio-2.0.so.0.4600.2
    [log.c:174] 7fd654d63000-7fd654d65000 rw-p 00000000 00:00 0 
    [log.c:174] 7fd654d65000-7fd654d68000 r-xp 00000000 fe:00 1848368                    /usr/lib/libgmodule-2.0.so.0.4600.2
    [log.c:174] 7fd654d68000-7fd654f67000 ---p 00003000 fe:00 1848368                    /usr/lib/libgmodule-2.0.so.0.4600.2
    [log.c:174] 7fd654f67000-7fd654f68000 r--p 00002000 fe:00 1848368                    /usr/lib/libgmodule-2.0.so.0.4600.2
    [log.c:174] 7fd654f68000-7fd654f69000 rw-p 00003000 fe:00 1848368                    /usr/lib/libgmodule-2.0.so.0.4600.2
    [log.c:174] 7fd654f69000-7fd654ffe000 r-xp 00000000 fe:00 1989114                    /usr/lib/mesa/libGL.so.1.2.0
    [log.c:174] 7fd654ffe000-7fd6551fd000 ---p 00095000 fe:00 1989114                    /usr/lib/mesa/libGL.so.1.2.0
    [log.c:174] 7fd6551fd000-7fd655200000 r--p 00094000 fe:00 1989114                    /usr/lib/mesa/libGL.so.1.2.0
    [log.c:174] 7fd655200000-7fd655201000 rw-p 00097000 fe:00 1989114                    /usr/lib/mesa/libGL.so.1.2.0
    [log.c:174] 7fd655201000-7fd655202000 rw-p 00000000 00:00 0 
    [log.c:174] 7fd655202000-7fd655217000 r-xp 00000000 fe:00 1841435                    /usr/lib/libz.so.1.2.8
    [log.c:174] 7fd655217000-7fd655416000 ---p 00015000 fe:00 1841435                    /usr/lib/libz.so.1.2.8
    [log.c:174] 7fd655416000-7fd655417000 r--p 00014000 fe:00 1841435                    /usr/lib/libz.so.1.2.8
    [log.c:174] 7fd655417000-7fd655418000 rw-p 00015000 fe:00 1841435                    /usr/lib/libz.so.1.2.8
    [log.c:174] 7fd655418000-7fd655429000 r-xp 00000000 fe:00 1861298                    /usr/lib/libXext.so.6.4.0
    [log.c:174] 7fd655429000-7fd655628000 ---p 00011000 fe:00 1861298                    /usr/lib/libXext.so.6.4.0
    [log.c:174] 7fd655628000-7fd655629000 r--p 00010000 fe:00 1861298                    /usr/lib/libXext.so.6.4.0
    [log.c:174] 7fd655629000-7fd65562a000 rw-p 00011000 fe:00 1861298                    /usr/lib/libXext.so.6.4.0
    [log.c:174] 7fd65562a000-7fd655633000 r-xp 00000000 fe:00 1862457                    /usr/lib/libXrender.so.1.3.0
    [log.c:174] 7fd655633000-7fd655832000 ---p 00009000 fe:00 1862457                    /usr/lib/libXrender.so.1.3.0
    [log.c:174] 7fd655832000-7fd655833000 r--p 00008000 fe:00 1862457                    /usr/lib/libXrender.so.1.3.0
    [log.c:174] 7fd655833000-7fd655834000 rw-p 00009000 fe:00 1862457                    /usr/lib/libXrender.so.1.3.0
    [log.c:174] 7fd655834000-7fd65583d000 r-xp 00000000 fe:00 1860315                    /usr/lib/libxcb-render.so.0.0.0
    [log.c:174] 7fd65583d000-7fd655a3c000 ---p 00009000 fe:00 1860315                    /usr/lib/libxcb-render.so.0.0.0
    [log.c:174] 7fd655a3c000-7fd655a3d000 r--p 00008000 fe:00 1860315                    /usr/lib/libxcb-render.so.0.0.0
    [log.c:174] 7fd655a3d000-7fd655a3e000 rw-p 00009000 fe:00 1860315                    /usr/lib/libxcb-render.so.0.0.0
    [log.c:174] 7fd655a3e000-7fd655a40000 r-xp 00000000 fe:00 1860349                    /usr/lib/libxcb-shm.so.0.0.0
    [log.c:174] 7fd655a40000-7fd655c40000 ---p 00002000 fe:00 1860349                    /usr/lib/libxcb-shm.so.0.0.0
    [log.c:174] 7fd655c40000-7fd655c41000 r--p 00002000 fe:00 1860349                    /usr/lib/libxcb-shm.so.0.0.0
    [log.c:174] 7fd655c41000-7fd655c42000 rw-p 00003000 fe:00 1860349                    /usr/lib/libxcb-shm.so.0.0.0
    [log.c:174] 7fd655c42000-7fd655c76000 r-xp 00000000 fe:00 1862444                    /usr/lib/libpng16.so.16.21.0
    [log.c:174] 7fd655c76000-7fd655e75000 ---p 00034000 fe:00 1862444                    /usr/lib/libpng16.so.16.21.0
    [log.c:174] 7fd655e75000-7fd655e76000 r--p 00033000 fe:00 1862444                    /usr/lib/libpng16.so.16.21.0
    [log.c:174] 7fd655e76000-7fd655e77000 rw-p 00034000 fe:00 1862444                    /usr/lib/libpng16.so.16.21.0
    [log.c:174] 7fd655e77000-7fd655f36000 r-xp 00000000 fe:00 1879484                    /usr/lib/libfreetype.so.6.12.2
    [log.c:174] 7fd655f36000-7fd656136000 ---p 000bf000 fe:00 1879484                    /usr/lib/libfreetype.so.6.12.2
    [log.c:174] 7fd656136000-7fd65613e000 r--p 000bf000 fe:00 1879484                    /usr/lib/libfreetype.so.6.12.2
    [log.c:174] 7fd65613e000-7fd65613f000 rw-p 000c7000 fe:00 1879484                    /usr/lib/libfreetype.so.6.12.2
    [log.c:174] 7fd65613f000-7fd65617d000 r-xp 00000000 fe:00 1862486                    /usr/lib/libfontconfig.so.1.9.0
    [log.c:174] 7fd65617d000-7fd65637c000 ---p 0003e000 fe:00 1862486                    /usr/lib/libfontconfig.so.1.9.0
    [log.c:174] 7fd65637c000-7fd65637e000 r--p 0003d000 fe:00 1862486                    /usr/lib/libfontconfig.so.1.9.0
    [log.c:174] 7fd65637e000-7fd656383000 rw-p 0003f000 fe:00 1862486                    /usr/lib/libfontconfig.so.1.9.0
    [log.c:174] 7fd656383000-7fd6563e2000 r-xp 00000000 fe:00 1872592                    /usr/lib/libharfbuzz.so.0.10200.3
    [log.c:174] 7fd6563e2000-7fd6565e2000 ---p 0005f000 fe:00 1872592                    /usr/lib/libharfbuzz.so.0.10200.3
    [log.c:174] 7fd6565e2000-7fd6565e3000 r--p 0005f000 fe:00 1872592                    /usr/lib/libharfbuzz.so.0.10200.3
    [log.c:174] 7fd6565e3000-7fd6565e4000 rw-p 00060000 fe:00 1872592                    /usr/lib/libharfbuzz.so.0.10200.3
    [log.c:174] 7fd6565e4000-7fd6565e5000 r-xp 00000000 fe:00 1848369                    /usr/lib/libgthread-2.0.so.0.4600.2
    [log.c:174] 7fd6565e5000-7fd6567e4000 ---p 00001000 fe:00 1848369                    /usr/lib/libgthread-2.0.so.0.4600.2
    [log.c:174] 7fd6567e4000-7fd6567e5000 r--p 00000000 fe:00 1848369                    /usr/lib/libgthread-2.0.so.0.4600.2
    [log.c:174] 7fd6567e5000-7fd6567e6000 rw-p 00001000 fe:00 1848369                    /usr/lib/libgthread-2.0.so.0.4600.2
    [log.c:174] 7fd6567e6000-7fd6567ee000 r-xp 00000000 fe:00 1862418                    /usr/lib/libthai.so.0.2.4
    [log.c:174] 7fd6567ee000-7fd6569ed000 ---p 00008000 fe:00 1862418                    /usr/lib/libthai.so.0.2.4
    [log.c:174] 7fd6569ed000-7fd6569ee000 r--p 00007000 fe:00 1862418                    /usr/lib/libthai.so.0.2.4
    [log.c:174] 7fd6569ee000-7fd6569ef000 rw-p 00008000 fe:00 1862418                    /usr/lib/libthai.so.0.2.4
    [log.c:174] 7fd6569ef000-7fd656a04000 r-xp 00000000 fe:00 1862766                    /usr/lib/libpangoft2-1.0.so.0.3900.0
    [log.c:174] 7fd656a04000-7fd656c03000 ---p 00015000 fe:00 1862766                    /usr/lib/libpangoft2-1.0.so.0.3900.0
    [log.c:174] 7fd656c03000-7fd656c04000 r--p 00014000 fe:00 1862766                    /usr/lib/libpangoft2-1.0.so.0.3900.0
    [log.c:174] 7fd656c04000-7fd656c05000 rw-p 00015000 fe:00 1862766                    /usr/lib/libpangoft2-1.0.so.0.3900.0
    [log.c:174] 7fd656c05000-7fd656c0e000 r-xp 00000000 fe:00 1882666                    /usr/lib/libwacom.so.2.4.5
    [log.c:174] 7fd656c0e000-7fd656e0d000 ---p 00009000 fe:00 1882666                    /usr/lib/libwacom.so.2.4.5
    [log.c:174] 7fd656e0d000-7fd656e0e000 r--p 00008000 fe:00 1882666                    /usr/lib/libwacom.so.2.4.5
    [log.c:174] 7fd656e0e000-7fd656e0f000 rw-p 00009000 fe:00 1882666                    /usr/lib/libwacom.so.2.4.5
    [log.c:174] 7fd656e0f000-7fd656e23000 r-xp 00000000 fe:00 1878987                    /usr/lib/libevdev.so.2.1.12
    [log.c:174] 7fd656e23000-7fd657022000 ---p 00014000 fe:00 1878987                    /usr/lib/libevdev.so.2.1.12
    [log.c:174] 7fd657022000-7fd657028000 r--p 00013000 fe:00 1878987                    /usr/lib/libevdev.so.2.1.12
    [log.c:174] 7fd657028000-7fd657029000 rw-p 00019000 fe:00 1878987                    /usr/lib/libevdev.so.2.1.12
    [log.c:174] 7fd657029000-7fd65702e000 r-xp 00000000 fe:00 1867087                    /usr/lib/libmtdev.so.1.0.0
    [log.c:174] 7fd65702e000-7fd65722d000 ---p 00005000 fe:00 1867087                    /usr/lib/libmtdev.so.1.0.0
    [log.c:174] 7fd65722d000-7fd65722e000 r--p 00004000 fe:00 1867087                    /usr/lib/libmtdev.so.1.0.0
    [log.c:174] 7fd65722e000-7fd65722f000 rw-p 00005000 fe:00 1867087                    /usr/lib/libmtdev.so.1.0.0
    [log.c:174] 7fd65722f000-7fd657236000 r-xp 00000000 fe:00 1838315                    /usr/lib/librt-2.23.so
    [log.c:174] 7fd657236000-7fd657435000 ---p 00007000 fe:00 1838315                    /usr/lib/librt-2.23.so
    [log.c:174] 7fd657435000-7fd657436000 r--p 00006000 fe:00 1838315                    /usr/lib/librt-2.23.so
    [log.c:174] 7fd657436000-7fd657437000 rw-p 00007000 fe:00 1838315                    /usr/lib/librt-2.23.so
    [log.c:174] 7fd657437000-7fd65743f000 r-xp 00000000 fe:00 1848294                    /usr/lib/libffi.so.6.0.4
    [log.c:174] 7fd65743f000-7fd65763e000 ---p 00008000 fe:00 1848294                    /usr/lib/libffi.so.6.0.4
    [log.c:174] 7fd65763e000-7fd65763f000 r--p 00007000 fe:00 1848294                    /usr/lib/libffi.so.6.0.4
    [log.c:174] 7fd65763f000-7fd657640000 rw-p 00008000 fe:00 1848294                    /usr/lib/libffi.so.6.0.4
    [log.c:174] 7fd657640000-7fd657658000 r-xp 00000000 fe:00 1838201                    /usr/lib/libpthread-2.23.so
    [log.c:174] 7fd657658000-7fd657857000 ---p 00018000 fe:00 1838201                    /usr/lib/libpthread-2.23.so
    [log.c:174] 7fd657857000-7fd657858000 r--p 00017000 fe:00 1838201                    /usr/lib/libpthread-2.23.so
    [log.c:174] 7fd657858000-7fd657859000 rw-p 00018000 fe:00 1838201                    /usr/lib/libpthread-2.23.so
    [log.c:174] 7fd657859000-7fd65785d000 rw-p 00000000 00:00 0 
    [log.c:174] 7fd65785d000-7fd6578ab000 r-xp 00000000 fe:00 1889806                    /usr/lib/libdbus-1.so.3.14.6
    [log.c:174] 7fd6578ab000-7fd657aab000 ---p 0004e000 fe:00 1889806                    /usr/lib/libdbus-1.so.3.14.6
    [log.c:174] 7fd657aab000-7fd657aac000 r--p 0004e000 fe:00 1889806                    /usr/lib/libdbus-1.so.3.14.6
    [log.c:174] 7fd657aac000-7fd657aad000 rw-p 0004f000 fe:00 1889806                    /usr/lib/libdbus-1.so.3.14.6
    [log.c:174] 7fd657aad000-7fd657aae000 rw-p 00000000 00:00 0 
    [log.c:174] 7fd657aae000-7fd657ab0000 r-xp 00000000 fe:00 1838311                    /usr/lib/libdl-2.23.so
    [log.c:174] 7fd657ab0000-7fd657cb0000 ---p 00002000 fe:00 1838311                    /usr/lib/libdl-2.23.so
    [log.c:174] 7fd657cb0000-7fd657cb1000 r--p 00002000 fe:00 1838311                    /usr/lib/libdl-2.23.so
    [log.c:174] 7fd657cb1000-7fd657cb2000 rw-p 00003000 fe:00 1838311                    /usr/lib/libdl-2.23.so
    [log.c:174] 7fd657cb2000-7fd657cbe000 r-xp 00000000 fe:00 1861529                    /usr/lib/libgbm.so.1.0.0
    [log.c:174] 7fd657cbe000-7fd657ebe000 ---p 0000c000 fe:00 1861529                    /usr/lib/libgbm.so.1.0.0
    [log.c:174] 7fd657ebe000-7fd657ebf000 r--p 0000c000 fe:00 1861529                    /usr/lib/libgbm.so.1.0.0
    [log.c:174] 7fd657ebf000-7fd657ec0000 rw-p 0000d000 fe:00 1861529                    /usr/lib/libgbm.so.1.0.0
    [log.c:174] 7fd657ec0000-7fd657ece000 r-xp 00000000 fe:00 1892348                    /usr/lib/libdrm.so.2.4.0
    [log.c:174] 7fd657ece000-7fd6580cd000 ---p 0000e000 fe:00 1892348                    /usr/lib/libdrm.so.2.4.0
    [log.c:174] 7fd6580cd000-7fd6580ce000 r--p 0000d000 fe:00 1892348                    /usr/lib/libdrm.so.2.4.0
    [log.c:174] 7fd6580ce000-7fd6580cf000 rw-p 0000e000 fe:00 1892348                    /usr/lib/libdrm.so.2.4.0
    [log.c:174] 7fd6580cf000-7fd6580fa000 r-xp 00000000 fe:00 1989115                    /usr/lib/mesa/libEGL.so.1.0.0
    [log.c:174] 7fd6580fa000-7fd6582fa000 ---p 0002b000 fe:00 1989115                    /usr/lib/mesa/libEGL.so.1.0.0
    [log.c:174] 7fd6582fa000-7fd6582fb000 r--p 0002b000 fe:00 1989115                    /usr/lib/mesa/libEGL.so.1.0.0
    [log.c:174] 7fd6582fb000-7fd6582fc000 rw-p 0002c000 fe:00 1989115                    /usr/lib/mesa/libEGL.so.1.0.0
    [log.c:174] 7fd6582fc000-7fd658303000 r-xp 00000000 fe:00 1989117                    /usr/lib/mesa/libGLESv2.so.2.0.0
    [log.c:174] 7fd658303000-7fd658502000 ---p 00007000 fe:00 1989117                    /usr/lib/mesa/libGLESv2.so.2.0.0
    [log.c:174] 7fd658502000-7fd658503000 r--p 00006000 fe:00 1989117                    /usr/lib/mesa/libGLESv2.so.2.0.0
    [log.c:174] 7fd658503000-7fd658504000 rw-p 00007000 fe:00 1989117                    /usr/lib/mesa/libGLESv2.so.2.0.0
    [log.c:174] 7fd658504000-7fd65850a000 r-xp 00000000 fe:00 1860311                    /usr/lib/libxcb-xfixes.so.0.0.0
    [log.c:174] 7fd65850a000-7fd65870a000 ---p 00006000 fe:00 1860311                    /usr/lib/libxcb-xfixes.so.0.0.0
    [log.c:174] 7fd65870a000-7fd65870b000 r--p 00006000 fe:00 1860311                    /usr/lib/libxcb-xfixes.so.0.0.0
    [log.c:174] 7fd65870b000-7fd65870c000 rw-p 00007000 fe:00 1860311                    /usr/lib/libxcb-xfixes.so.0.0.0
    [log.c:174] 7fd65870c000-7fd658710000 r-xp 00000000 fe:00 1862363                    /usr/lib/libxcb-image.so.0.0.0
    [log.c:174] 7fd658710000-7fd65890f000 ---p 00004000 fe:00 1862363                    /usr/lib/libxcb-image.so.0.0.0
    [log.c:174] 7fd65890f000-7fd658910000 r--p 00003000 fe:00 1862363                    /usr/lib/libxcb-image.so.0.0.0
    [log.c:174] 7fd658910000-7fd658911000 rw-p 00004000 fe:00 1862363                    /usr/lib/libxcb-image.so.0.0.0
    [log.c:174] 7fd658911000-7fd65892c000 r-xp 00000000 fe:00 1860391                    /usr/lib/libxcb-xkb.so.1.0.0
    [log.c:174] 7fd65892c000-7fd658b2b000 ---p 0001b000 fe:00 1860391                    /usr/lib/libxcb-xkb.so.1.0.0
    [log.c:174] 7fd658b2b000-7fd658b2c000 r--p 0001a000 fe:00 1860391                    /usr/lib/libxcb-xkb.so.1.0.0
    [log.c:174] 7fd658b2c000-7fd658b2d000 rw-p 0001b000 fe:00 1860391                    /usr/lib/libxcb-xkb.so.1.0.0
    [log.c:174] 7fd658b2d000-7fd658b2f000 r-xp 00000000 fe:00 1860361                    /usr/lib/libxcb-composite.so.0.0.0
    [log.c:174] 7fd658b2f000-7fd658d2f000 ---p 00002000 fe:00 1860361                    /usr/lib/libxcb-composite.so.0.0.0
    [log.c:174] 7fd658d2f000-7fd658d30000 r--p 00002000 fe:00 1860361                    /usr/lib/libxcb-composite.so.0.0.0
    [log.c:174] 7fd658d30000-7fd658d31000 rw-p 00003000 fe:00 1860361                    /usr/lib/libxcb-composite.so.0.0.0
    [log.c:174] 7fd658d31000-7fd658d52000 r-xp 00000000 fe:00 1860337                    /usr/lib/libxcb.so.1.1.0
    [log.c:174] 7fd658d52000-7fd658f52000 ---p 00021000 fe:00 1860337                    /usr/lib/libxcb.so.1.1.0
    [log.c:174] 7fd658f52000-7fd658f53000 r--p 00021000 fe:00 1860337                    /usr/lib/libxcb.so.1.1.0
    [log.c:174] 7fd658f53000-7fd658f54000 rw-p 00022000 fe:00 1860337                    /usr/lib/libxcb.so.1.1.0
    [log.c:174] 7fd658f54000-7fd658f55000 r-xp 00000000 fe:00 1860413                    /usr/lib/libX11-xcb.so.1.0.0
    [log.c:174] 7fd658f55000-7fd659154000 ---p 00001000 fe:00 1860413                    /usr/lib/libX11-xcb.so.1.0.0
    [log.c:174] 7fd659154000-7fd659155000 r--p 00000000 fe:00 1860413                    /usr/lib/libX11-xcb.so.1.0.0
    [log.c:174] 7fd659155000-7fd659156000 rw-p 00001000 fe:00 1860413                    /usr/lib/libX11-xcb.so.1.0.0
    [log.c:174] 7fd659156000-7fd659291000 r-xp 00000000 fe:00 1860412                    /usr/lib/libX11.so.6.3.0
    [log.c:174] 7fd659291000-7fd659490000 ---p 0013b000 fe:00 1860412                    /usr/lib/libX11.so.6.3.0
    [log.c:174] 7fd659490000-7fd659492000 r--p 0013a000 fe:00 1860412                    /usr/lib/libX11.so.6.3.0
    [log.c:174] 7fd659492000-7fd659497000 rw-p 0013c000 fe:00 1860412                    /usr/lib/libX11.so.6.3.0
    [log.c:174] 7fd659497000-7fd659498000 rw-p 00000000 00:00 0 
    [log.c:174] 7fd659498000-7fd659537000 r-xp 00000000 fe:00 1861592                    /usr/lib/libpixman-1.so.0.34.0
    [log.c:174] 7fd659537000-7fd659737000 ---p 0009f000 fe:00 1861592                    /usr/lib/libpixman-1.so.0.34.0
    [log.c:174] 7fd659737000-7fd65973f000 r--p 0009f000 fe:00 1861592                    /usr/lib/libpixman-1.so.0.34.0
    [log.c:174] 7fd65973f000-7fd659740000 rw-p 000a7000 fe:00 1861592                    /usr/lib/libpixman-1.so.0.34.0
    [log.c:174] 7fd659740000-7fd6598d8000 r-xp 00000000 fe:00 1838236                    /usr/lib/libc-2.23.so
    [log.c:174] 7fd6598d8000-7fd659ad7000 ---p 00198000 fe:00 1838236                    /usr/lib/libc-2.23.so
    [log.c:174] 7fd659ad7000-7fd659adb000 r--p 00197000 fe:00 1838236                    /usr/lib/libc-2.23.so
    [log.c:174] 7fd659adb000-7fd659add000 rw-p 0019b000 fe:00 1838236                    /usr/lib/libc-2.23.so
    [log.c:174] 7fd659add000-7fd659ae1000 rw-p 00000000 00:00 0 
    [log.c:174] 7fd659ae1000-7fd659b05000 r-xp 00000000 fe:00 1863535                    /usr/lib/libgdk_pixbuf-2.0.so.0.3200.3
    [log.c:174] 7fd659b05000-7fd659d05000 ---p 00024000 fe:00 1863535                    /usr/lib/libgdk_pixbuf-2.0.so.0.3200.3
    [log.c:174] 7fd659d05000-7fd659d06000 r--p 00024000 fe:00 1863535                    /usr/lib/libgdk_pixbuf-2.0.so.0.3200.3
    [log.c:174] 7fd659d06000-7fd659d07000 rw-p 00025000 fe:00 1863535                    /usr/lib/libgdk_pixbuf-2.0.so.0.3200.3
    [log.c:174] 7fd659d07000-7fd659e0a000 r-xp 00000000 fe:00 1838312                    /usr/lib/libm-2.23.so
    [log.c:174] 7fd659e0a000-7fd65a00a000 ---p 00103000 fe:00 1838312                    /usr/lib/libm-2.23.so
    [log.c:174] 7fd65a00a000-7fd65a00b000 r--p 00103000 fe:00 1838312                    /usr/lib/libm-2.23.so
    [log.c:174] 7fd65a00b000-7fd65a00c000 rw-p 00104000 fe:00 1838312                    /usr/lib/libm-2.23.so
    [log.c:174] 7fd65a00c000-7fd65a135000 r-xp 00000000 fe:00 1862721                    /usr/lib/libcairo.so.2.11400.4
    [log.c:174] 7fd65a135000-7fd65a335000 ---p 00129000 fe:00 1862721                    /usr/lib/libcairo.so.2.11400.4
    [log.c:174] 7fd65a335000-7fd65a338000 r--p 00129000 fe:00 1862721                    /usr/lib/libcairo.so.2.11400.4
    [log.c:174] 7fd65a338000-7fd65a33a000 rw-p 0012c000 fe:00 1862721                    /usr/lib/libcairo.so.2.11400.4
    [log.c:174] 7fd65a33a000-7fd65a33b000 rw-p 00000000 00:00 0 
    [log.c:174] 7fd65a33b000-7fd65a447000 r-xp 00000000 fe:00 1848367                    /usr/lib/libglib-2.0.so.0.4600.2
    [log.c:174] 7fd65a447000-7fd65a646000 ---p 0010c000 fe:00 1848367                    /usr/lib/libglib-2.0.so.0.4600.2
    [log.c:174] 7fd65a646000-7fd65a647000 r--p 0010b000 fe:00 1848367                    /usr/lib/libglib-2.0.so.0.4600.2
    [log.c:174] 7fd65a647000-7fd65a648000 rw-p 0010c000 fe:00 1848367                    /usr/lib/libglib-2.0.so.0.4600.2
    [log.c:174] 7fd65a648000-7fd65a649000 rw-p 00000000 00:00 0 
    [log.c:174] 7fd65a649000-7fd65a69a000 r-xp 00000000 fe:00 1848370                    /usr/lib/libgobject-2.0.so.0.4600.2
    [log.c:174] 7fd65a69a000-7fd65a899000 ---p 00051000 fe:00 1848370                    /usr/lib/libgobject-2.0.so.0.4600.2
    [log.c:174] 7fd65a899000-7fd65a89a000 r--p 00050000 fe:00 1848370                    /usr/lib/libgobject-2.0.so.0.4600.2
    [log.c:174] 7fd65a89a000-7fd65a89b000 rw-p 00051000 fe:00 1848370                    /usr/lib/libgobject-2.0.so.0.4600.2
    [log.c:174] 7fd65a89b000-7fd65a8e4000 r-xp 00000000 fe:00 1862761                    /usr/lib/libpango-1.0.so.0.3900.0
    [log.c:174] 7fd65a8e4000-7fd65aae4000 ---p 00049000 fe:00 1862761                    /usr/lib/libpango-1.0.so.0.3900.0
    [log.c:174] 7fd65aae4000-7fd65aae6000 r--p 00049000 fe:00 1862761                    /usr/lib/libpango-1.0.so.0.3900.0
    [log.c:174] 7fd65aae6000-7fd65aae7000 rw-p 0004b000 fe:00 1862761                    /usr/lib/libpango-1.0.so.0.3900.0
    [log.c:174] 7fd65aae7000-7fd65aaf3000 r-xp 00000000 fe:00 1862768                    /usr/lib/libpangocairo-1.0.so.0.3900.0
    [log.c:174] 7fd65aaf3000-7fd65acf2000 ---p 0000c000 fe:00 1862768                    /usr/lib/libpangocairo-1.0.so.0.3900.0
    [log.c:174] 7fd65acf2000-7fd65acf3000 r--p 0000b000 fe:00 1862768                    /usr/lib/libpangocairo-1.0.so.0.3900.0
    [log.c:174] 7fd65acf3000-7fd65acf4000 rw-p 0000c000 fe:00 1862768                    /usr/lib/libpangocairo-1.0.so.0.3900.0
    [log.c:174] 7fd65acf4000-7fd65ad1d000 r-xp 00000000 fe:00 1853370                    /usr/lib/libinput.so.10.7.4
    [log.c:174] 7fd65ad1d000-7fd65af1c000 ---p 00029000 fe:00 1853370                    /usr/lib/libinput.so.10.7.4
    [log.c:174] 7fd65af1c000-7fd65af1d000 r--p 00028000 fe:00 1853370                    /usr/lib/libinput.so.10.7.4
    [log.c:174] 7fd65af1d000-7fd65af1e000 rw-p 00029000 fe:00 1853370                    /usr/lib/libinput.so.10.7.4
    [log.c:174] 7fd65af1e000-7fd65af2e000 r-xp 00000000 fe:00 1857922                    /usr/lib/libwayland-server.so.0.1.0
    [log.c:174] 7fd65af2e000-7fd65b12d000 ---p 00010000 fe:00 1857922                    /usr/lib/libwayland-server.so.0.1.0
    [log.c:174] 7fd65b12d000-7fd65b12f000 r--p 0000f000 fe:00 1857922                    /usr/lib/libwayland-server.so.0.1.0
    [log.c:174] 7fd65b12f000-7fd65b130000 rw-p 00011000 fe:00 1857922                    /usr/lib/libwayland-server.so.0.1.0
    [log.c:174] 7fd65b130000-7fd65b13a000 r-xp 00000000 fe:00 1864884                    /usr/lib/libjson-c.so.2.0.1
    [log.c:174] 7fd65b13a000-7fd65b339000 ---p 0000a000 fe:00 1864884                    /usr/lib/libjson-c.so.2.0.1
    [log.c:174] 7fd65b339000-7fd65b33a000 r--p 00009000 fe:00 1864884                    /usr/lib/libjson-c.so.2.0.1
    [log.c:174] 7fd65b33a000-7fd65b33b000 rw-p 0000a000 fe:00 1864884                    /usr/lib/libjson-c.so.2.0.1
    [log.c:174] 7fd65b33b000-7fd65b3aa000 r-xp 00000000 fe:00 1846593                    /usr/lib/libpcre.so.1.2.6
    [log.c:174] 7fd65b3aa000-7fd65b5a9000 ---p 0006f000 fe:00 1846593                    /usr/lib/libpcre.so.1.2.6
    [log.c:174] 7fd65b5a9000-7fd65b5aa000 r--p 0006e000 fe:00 1846593                    /usr/lib/libpcre.so.1.2.6
    [log.c:174] 7fd65b5aa000-7fd65b5ab000 rw-p 0006f000 fe:00 1846593                    /usr/lib/libpcre.so.1.2.6
    [log.c:174] 7fd65b5ab000-7fd65b5e8000 r-xp 00000000 fe:00 1862812                    /usr/lib/libxkbcommon.so.0.0.0
    [log.c:174] 7fd65b5e8000-7fd65b7e8000 ---p 0003d000 fe:00 1862812                    /usr/lib/libxkbcommon.so.0.0.0
    [log.c:174] 7fd65b7e8000-7fd65b7ea000 r--p 0003d000 fe:00 1862812                    /usr/lib/libxkbcommon.so.0.0.0
    [log.c:174] 7fd65b7ea000-7fd65b7eb000 rw-p 0003f000 fe:00 1862812                    /usr/lib/libxkbcommon.so.0.0.0
    [log.c:174] 7fd65b7eb000-7fd65b827000 r-xp 00000000 fe:00 1845386                    /usr/lib/libwlc.so.0.0.1
    [log.c:174] 7fd65b827000-7fd65ba27000 ---p 0003c000 fe:00 1845386                    /usr/lib/libwlc.so.0.0.1
    [log.c:174] 7fd65ba27000-7fd65ba29000 r--p 0003c000 fe:00 1845386                    /usr/lib/libwlc.so.0.0.1
    [log.c:174] 7fd65ba29000-7fd65ba2a000 rw-p 0003e000 fe:00 1845386                    /usr/lib/libwlc.so.0.0.1
    [log.c:174] 7fd65ba2a000-7fd65ba2b000 rw-p 00000000 00:00 0 
    [log.c:174] 7fd65ba2b000-7fd65ba4e000 r-xp 00000000 fe:00 1838232                    /usr/lib/ld-2.23.so
    [log.c:174] 7fd65ba4e000-7fd65ba4f000 rw-s 00000000 00:05 299598                     /drm mm object (deleted)
    [log.c:174] 7fd65ba4f000-7fd65ba50000 rw-s 100001000 00:06 13742                     /dev/dri/card0
    [log.c:174] 7fd65ba50000-7fd65ba51000 rw-s 00000000 00:05 299587                     /drm mm object (deleted)
    [log.c:174] 7fd65ba51000-7fd65ba52000 rw-s 00000000 00:05 299585                     /drm mm object (deleted)
    [log.c:174] 7fd65ba52000-7fd65ba53000 rw-s 00000000 00:05 299577                     /drm mm object (deleted)
    [log.c:174] 7fd65ba53000-7fd65ba54000 rw-s 100000000 00:06 13742                     /dev/dri/card0
    [log.c:174] 7fd65ba54000-7fd65ba5c000 rw-s 00000000 00:05 299575                     /drm mm object (deleted)
    [log.c:174] 7fd65ba5c000-7fd65baa9000 r--p 00000000 fe:00 528662                     /usr/share/fonts/ttf-liberation-ib/LiberationMono-Regular.ttf
    [log.c:174] 7fd65baa9000-7fd65baae000 r--s 00000000 fe:00 143276                     /var/cache/fontconfig/6d78b720ace5d3bb3fea22ee600d50bf-le64.cache-7
    [log.c:174] 7fd65baae000-7fd65bab4000 r--s 00000000 fe:00 143275                     /var/cache/fontconfig/558cfb7b00557d1eaf02482719df5bbf-le64.cache-7
    [log.c:174] 7fd65bab4000-7fd65bad2000 r--s 00000000 fe:00 143274                     /var/cache/fontconfig/2aeaefbbe8fee9acf2c287bc92ac119e-le64.cache-7
    [log.c:174] 7fd65bad2000-7fd65bad7000 r--s 00000000 fe:00 143273                     /var/cache/fontconfig/b987be13d9555b03a6cd440467233128-le64.cache-7
    [log.c:174] 7fd65bad7000-7fd65bade000 r--s 00000000 fe:00 143272                     /var/cache/fontconfig/05c6939ffcab4ece42a69ecb7571bcfc-le64.cache-7
    [log.c:174] 7fd65bade000-7fd65bae2000 r--s 00000000 fe:00 143271                     /var/cache/fontconfig/15307eb168539589f690be8d29ed1550-le64.cache-7
    [log.c:174] 7fd65bae2000-7fd65baed000 r--s 00000000 fe:00 143270                     /var/cache/fontconfig/81afcec12fadd71c4e03e0a437247ada-le64.cache-7
    [log.c:174] 7fd65baed000-7fd65baf7000 r--s 00000000 fe:00 143269                     /var/cache/fontconfig/eec1da544b6dea79174198a09a665442-le64.cache-7
    [log.c:174] 7fd65baf7000-7fd65baf8000 r--s 00000000 fe:00 143268                     /var/cache/fontconfig/dc14ea377282698501b78c9ee88ac8f5-le64.cache-7
    [log.c:174] 7fd65baf8000-7fd65bb18000 r--s 00000000 fe:00 143282                     /var/cache/fontconfig/8d4af663993b81a124ee82e610bb31f9-le64.cache-7
    [log.c:174] 7fd65bb18000-7fd65bb45000 r--s 00000000 fe:00 143248                     /var/cache/fontconfig/f6b893a7224233d96cb72fd88691c0b4-le64.cache-7
    [log.c:174] 7fd65bb45000-7fd65bb5e000 rw-p 00000000 00:00 0 
    [log.c:174] 7fd65bb5e000-7fd65bbdf000 r-xp 00000000 fe:00 1846522                    /usr/lib/libsystemd.so.0.14.0
    [log.c:174] 7fd65bbdf000-7fd65bbe2000 r--p 00080000 fe:00 1846522                    /usr/lib/libsystemd.so.0.14.0
    [log.c:174] 7fd65bbe2000-7fd65bbe3000 rw-p 00083000 fe:00 1846522                    /usr/lib/libsystemd.so.0.14.0
    [log.c:174] 7fd65bbe3000-7fd65bbe8000 rw-p 00000000 00:00 0 
    [log.c:174] 7fd65bbe8000-7fd65bc06000 r-xp 00000000 fe:00 1846523                    /usr/lib/libudev.so.1.6.4
    [log.c:174] 7fd65bc06000-7fd65bc07000 r--p 0001d000 fe:00 1846523                    /usr/lib/libudev.so.1.6.4
    [log.c:174] 7fd65bc07000-7fd65bc08000 rw-p 0001e000 fe:00 1846523                    /usr/lib/libudev.so.1.6.4
    [log.c:174] 7fd65bc08000-7fd65bc0e000 rw-p 00000000 00:00 0 
    [log.c:174] 7fd65bc0e000-7fd65bc11000 r--s 00000000 fe:00 143267                     /var/cache/fontconfig/d4119bf364b3cb6badea4cd5ba92dc93-le64.cache-7
    [log.c:174] 7fd65bc11000-7fd65bc1a000 r--s 00000000 fe:00 135961                     /var/cache/fontconfig/f349e9996a5320f6dd491cedd2b1f964-le64.cache-7
    [log.c:174] 7fd65bc1a000-7fd65bc42000 rw-p 00000000 00:00 0 
    [log.c:174] 7fd65bc42000-7fd65bc4e000 rw-s 00000000 00:2c 274394                     /run/user/1000/wlc-shared-UyYh3N (deleted)
    [log.c:174] 7fd65bc4e000-7fd65bc4f000 r--p 00023000 fe:00 1838232                    /usr/lib/ld-2.23.so
    [log.c:174] 7fd65bc4f000-7fd65bc50000 rw-p 00024000 fe:00 1838232                    /usr/lib/ld-2.23.so
    [log.c:174] 7fd65bc50000-7fd65bc51000 rw-p 00000000 00:00 0 
    [log.c:174] 7ffd5dc08000-7ffd5dc29000 rw-p 00000000 00:00 0                          [stack]
    [log.c:174] 7ffd5dc8c000-7ffd5dc8e000 r--p 00000000 00:00 0                          [vvar]
    [log.c:174] 7ffd5dc8e000-7ffd5dc90000 r-xp 00000000 00:00 0                          [vdso]
    [log.c:174] ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
    [log.c:174]

mikkeloscar commented:
> Isn't there supposed to be a different border for windows that are focused, but on an output that is not focused?

The border should be light grey with white font by default. But I forgot to test on multiple outputs (should work with multiple containers on one output). I'll fix this.

> My font is incorrect now for swaybar. Oddly it's correct for the window titles.

I refactored wayland/pango.c to use it with the borders, might have messed something up there.

> Another issue: you can see the border through transparent surfaces.

Forgot about this, will fix.

> I get a crash on this branch when initiating a download in firefox.

Could be related to what I removed in `handle_view_geometry_request`.

SirCmpwn commented:
>The border should be light grey with white font by default. But I forgot to test on multiple outputs (should work with multiple containers on one output). I'll fix this.

It appears to work for me with multiple containers on one output.

mikkeloscar commented on sway/border.c:
```diff
@@ -0,0 +1,253 @@
+	if (view->border) {
+		free(view->border);
+		view->border = NULL;
+	}
```
The border is freed here before it's calloced in `create_border_buffer`.

mikkeloscar commented:
Latest commit fixes the problems with gnome-calculator and firefox crashing. It also exposes #505 more clearly since the file dialog on firefox downloads is shifted off the expected geometry. I won't attempt to fix that issue in this PR.

SirCmpwn commented:
Woohoo!
